content/spire/try/oidc-federation-aws.md

```diff
@@ -0,0 +1,415 @@
+
+To get the external IP address required for the A record, you will need to proceed through the steps in this tutorial until you start the spire-oidc service and verify the external IP address in the section [Verify That spire-oidc has an External Service Address](#verify-that-spireoidc-has-an-external-service-address). The DNS A record should take the following form:
+
+```console
+MY_DISCOVERY_DOMAIN          A        EXTERNAL-IP for spire-oidc service
+```
+
+For example:
+```console
+oidc-discovery.example.org   A        93.184.216.34
+```
+
+{{< info >}}
+As with any change to DNS, it will take minutes or hours for the new A record to propagate to DNS servers. This tutorial will not work until the A record is propagated. Once the A record is propagated, you should be able to view the discovery document at https://MY_DISCOVERY_DOMAIN/.well-known/openid-configuration .
+{{< /info >}}
+
+Although this tutorial should contain all the info you need, the SPIRE OIDC Discovery Provider configuration file format is described on [GitHub](https://github.com/spiffe/spire/tree/master/support/oidc-discovery-provider).
+
+This tutorial uses Let's Encrypt to configure a certificate for the OIDC Discovery Provider domain. But other than editing the oidc-dp-configmap.yaml file as described in the previous section, no additional setup steps are required for Let's Encrypt.
+
+# Part One: Configure SPIRE Components
+
+In the first part of this procedure, you configure the SPIRE components. In the second part, you configure the AWS components. You test the connection in the third part.
+
+## Download Kubernetes YAML Files for this Tutorial
+
+To get the YAML files required for this tutorial, clone https://github.com/spiffe/spire-tutorials. The files for this tutorial are in the **k8s/oidc-aws** directory.
+
+## Replace Placeholder Strings in YAML Files
+
+The following strings in the YAML files must be substituted for values specific to your environment. Each location where you must make a change has been marked with `TODO:` in the YAML files.
+
+| String | Description | Files to Change |
+| --- | --- | --- |
+| MY\_EMAIL\_ADDRESS | The terms of service for the Let's Encrypt certificate authority requires that you specify a valid email. Let's Encrypt is used to configure a certificate for the OIDC Discovery Domain (see the [Discovery Provider](#required-dns-a-record-for-the-oidc-discovery-provider-ip-address) info above). Example value: user@example.org | oidc-dp-configmap.yaml (1 instance) |
+| MY\_DISCOVERY\_DOMAIN | You must configure a public DNS A record for the IP address of the OIDC Discovery Provider. See the next section for details. Example value: `oidc-discovery.example.org` | ingress.yaml (2 instances), oidc-dp-configmap.yaml (1 instance), server-configmap.yaml (1 instance) |
```

Comment:
This should focus on just what the user needs to do at this step - which is to decide on a custom sub domain name and to set it in the correct value in this file.

Reply:
Agreed, done.